worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -191,32 +237,44 @@ private void deleteExistingDirectory(Path outputDirPath) {
         }
     }
 
+    /**
+     * Return the list of attested contracts for a job. If a contract was specified in the job, just return that
+     * after checking to make sure the sponsor has access to the contract, otherwise, search for all the contracts
+     * for the sponsor
+     *
+     * @param job - the submitted job
+     * @return the list of contracts (all or only 1 if the contract was specified in the job).
+     */
     private List<Contract> getAttestedContracts(Job job) {
 
-        // when the job is submitted for a specific contract, process the export for only that contract.
+        // Get the aggregated attested Contracts for the sponsor
+        final Sponsor sponsor = job.getUser().getSponsor();
+        final List<Contract> attestedContracts = sponsor.getAggregatedAttestedContracts();
+
+        // If a contract was specified for request, make sure the sponsor can access the contract and then return only it
         final Contract jobSpecificContract = job.getContract();
         if (jobSpecificContract != null && jobSpecificContract.getAttestedOn() != null) {
-            log.info("Job [{}] submitted for a specific attested contract [{}] ", jobSpecificContract.getContractNumber());
+            boolean ownsContract = attestedContracts.stream()
+                    .anyMatch(c -> jobSpecificContract.getContractNumber().equalsIgnoreCase(c.getContractNumber()));
+            if (!ownsContract) {
+                log.info("Job [{}] submitted for a specific attested contract [{}] that the sponsor [{}] does not own",
+                        job.getJobUuid(), jobSpecificContract.getContractNumber(), sponsor.getOrgName());
+            }
+            log.info("Job [{}] submitted for a specific attested contract [{}] ", job.getJobUuid(), jobSpecificContract.getContractNumber());
```

Comment:
We are already guarding against this in `gov.cms.ab2d.common.service.JobServiceImpl#createJob(java.lang.String, java.lang.String, java.lang.String)`. Do we need this additional check?

Reply:
It's not expensive, why not? If the other class gets refactored, it might disappear and it's a pretty big security hole if PDPs get access to contracts that aren't theirs.